docs/integrations/optuna.rst

```diff
@@ -3,150 +3,214 @@
 
+Open the link in a new tab. The charts will currently be empty, but keep the window open. You will be able to see live metrics once logging starts.
 
-Monitor your Optuna training in Neptune
----------------------------------------
-Now you can watch your Optuna hyperparameter optimization in Neptune!
+Step 3: Create the Neptune Callback
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. code-block:: python3
 
-Check out this |example experiment|.
+   import neptunecontrib.monitoring.optuna as opt_utils
 
-.. image:: ../_static/images/integrations/optuna_monitoring.gif
-   :target: ../_static/images/integrations/optuna_monitoring.gif
-   :alt: Optuna monitoring in Neptune
+   neptune_callback = opt_utils.NeptuneCallback()
 
-Full script
------------
+Step 4: Running Optuna with the Neptune callback
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Pass the ``neptune_callback`` as a callback to ``study.optimize()`` to monitor the metrics and parameters checked at each run.
 
 .. code-block:: python3
 
-    import lightgbm as lgb
-    import optuna
-    from sklearn.datasets import load_breast_cancer
-    from sklearn.metrics import roc_auc_score
-    from sklearn.model_selection import train_test_split
+   study = optuna.create_study(direction='maximize')
+   study.optimize(objective, n_trials=100, callbacks=[neptune_callback])
 
-    import neptune
-    import neptunecontrib.monitoring.optuna as optuna_utils
+Step 5: Monitor your Optuna training in Neptune
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Now you can switch to the Neptune tab which you had opened previously to watch the optimization live!
 
-    neptune.init(api_token='ANONYMOUS',
-                 project_qualified_name='shared/showroom')
+Check out this |example experiment|.
 
-    neptune.create_experiment('optuna-sweep')
-    neptune_callback = optuna_utils.NeptuneCallback()
+|optuna-basic-logs|
 
-    def objective(trial):
-        data, target = load_breast_cancer(return_X_y=True)
-        train_x, test_x, train_y, test_y = train_test_split(data, target, test_size=0.25)
-        dtrain = lgb.Dataset(train_x, label=train_y)
+.. _advanced-options:
 
-        param = {
-            'objective': 'binary',
-            'metric': 'binary_logloss',
-            'num_leaves': trial.suggest_int('num_leaves', 2, 256),
-            'feature_fraction': trial.suggest_uniform('feature_fraction', 0.4, 1.0),
-            'bagging_fraction': trial.suggest_uniform('bagging_fraction', 0.4, 1.0),
-            'min_child_samples': trial.suggest_int('min_child_samples', 5, 100),
-        }
+Advanced Options
+----------------
 
-        gbm = lgb.train(param, dtrain)
-        preds = gbm.predict(test_x)
-        accuracy = roc_auc_score(test_y, preds)
-        return accuracy
+While creating the Neptune Callback, you can set ``log_study=True`` and ``log_charts=True`` to log interactive charts from ``optuna.visualization`` and the study object itself after every iteration.
```

Comment:
I really think we should have.

Log charts and study during training (every iteration)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

CODE

It is not explicitly said anywhere that to log those charts you need to do this. 

Log charts and study after training 
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^